handler.js

```diff
@@ -127,11 +142,28 @@ HyperbahnHandler.prototype.type = 'hyperbahn.advertisement-handler';
 */
 HyperbahnHandler.prototype.handleAdvertise =
 function handleAdvertise(handler, req, arg2, arg3, cb) {
-    handler.sendRelays(req, arg2, arg3, 'relay-ad', cb);
+    var services = arg3.services;
+    var fanoutServices = [];
+
+    for (var i = 0; i < services.length; i++) {
```

Comment:
Could we do this logic in `#sendRelays` instead? There's already a loop over the host ports there that adds them to a new data structure, would be nice to only do that once.